tests/test_single_topology_combined.py

```diff
@@ -0,0 +1,381 @@
+# test code for combining host guest systems
+import matplotlib.pyplot as plt
+import numpy as np
+import pymbar
+import pytest
+
+from timemachine.constants import BOLTZ
+from timemachine.fe import pdb_writer
+from timemachine.fe.system import VacuumSystem
+from timemachine.fe.utils import get_romol_conf, to_md_units
+from timemachine.ff.handlers import openmm_deserializer
+from timemachine.lib import LangevinIntegrator, MonteCarloBarostat, custom_ops, potentials
+from timemachine.md import builders, minimizer
+from timemachine.md.barostat.utils import get_bond_list, get_group_indices
+from timemachine.testsystems.dhfr import get_dhfr_system
+from timemachine.testsystems.relative import get_hif2a_ligand_pair_single_topology
+
+
+def convert_bps_into_system(bps):
+
+    system = VacuumSystem(None, None, None, None, None, None)
+
+    for pot in bps:
+        if isinstance(pot, potentials.HarmonicBond):
+            system.bond = pot
+        elif isinstance(pot, potentials.HarmonicAngle):
+            system.angle = pot
+        elif isinstance(pot, potentials.PeriodicTorsion):
+            system.torsion = pot
+        elif isinstance(pot, potentials.Nonbonded):
+            system.nonbonded = pot
+        else:
+            assert 0, "Unknown potential"
+
+    return system
+
+
+def setup_potentials(host_bps, st, lamb):
+
+    host_system = VacuumSystem(None, None, None, None, None, None)
+
+    for pot in host_bps:
+        if isinstance(pot, potentials.HarmonicBond):
+            host_system.bond = pot
+        elif isinstance(pot, potentials.HarmonicAngle):
+            host_system.angle = pot
+        elif isinstance(pot, potentials.PeriodicTorsion):
+            host_system.torsion = pot
+        elif isinstance(pot, potentials.Nonbonded):
+            host_system.nonbonded = pot
+        else:
+            assert 0, "Unknown potential"
+
+    return st.combine_with_host(host_system, lamb=lamb)
+
+
+def sample(bound_impls, x0, v0, box0, lamb, group_idxs, seed, n_frames=500, burn_in=10000):
+
+    dt = 1e-3
+    total_num_atoms = x0.shape[0]
+
+    temperature = 300.0
+    friction = 1.0
+    masses = np.ones(total_num_atoms) * 5  # fixme
```

Comment:
oops - forgot to fix this!

Reply:
ah I remember why I did it this way. I'm not sure how to set the masses here for core atoms. We have several choices:

1) For core atoms, at lambda = 0, use mol_a's masses. at lambda=1 use mol_b's masses. At 0<lambda<1 linearly interpolate
2) Use only mol_a or only mol_b 
3) Optimize masses at each lambda window based on the interpolated bond and angle parameters.

@maxentile @jkausrelay thoughts?

Reply:
in light of not wanting to hold this up, I'll implement 1) now and re-visit this later

Reply:
I'm going to implement 4) take the max

Reply:
4) sounds safer than 1-3) for now, to avoid introducing any complications in HREX etc. from having different masses in different lambda windows.

In a later pass, 3) might buy some efficiency improvement